### PR TITLE
Display Python SDK version

### LINKED ISSUE #1034: Python component-version is not displayed in logs
It should appear [here](https://github.com/open-telemetry/opentelemetry-lambda/actions/runs/7133459715/job/19426262691#step:1:30).

### PR DESCRIPTION
**Pull request by ocelotl**
Fixes #1034

### CODE REVIEW

File: python/src/otel/otel_sdk/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+opentelemetry-sdk==1.21.0
```
Comment (ocelotl) [inline python/src/otel/otel_sdk/requirements.txt:1]: Dependabot should update this with every new release.

File: .github/workflows/release-layer-python.yml [hunk 2]
```diff
@@ -14,30 +14,40 @@ jobs:
...
+        shell: bash
+
+      - name: Display PYTHON_OPENTELEMETRY_SDK_VERSION
+        run: echo "PYTHON_OPENTELEMETRY_SDK_VERSION=${{ env.PYTHON_OPENTELEMETRY_SDK_VERSION }}"
+        shell: bash
+
+      - name: Save Python SDK Version
+        id: save-python-sdk-version
+        run: |
+          echo "PYTHON_OPENTELEMETRY_SDK_VERSION=$PYTHON_OPENTELEMETRY_SDK_VERSION" >> $GITHUB_OUTPUT
```
Comment (tylerbenson) [inline .github/workflows/release-layer-python.yml]: Can these three steps be combined into one?  I think I'd prefer that.

File: python/src/otel/otel_sdk/requirements-nodeps.txt [hunk 3]
```diff
@@ -27,4 +27,3 @@ opentelemetry-instrumentation-sqlite3==0.42b0
 opentelemetry-instrumentation-starlette==0.42b0
 opentelemetry-instrumentation-tornado==0.42b0
 opentelemetry-instrumentation-wsgi==0.42b0
-
```
Comment (tylerbenson) [inline python/src/otel/otel_sdk/requirements-nodeps.txt:30]: What about the `0.42b0` portion of the version... does that matter?  What is the difference?

File: .github/workflows/release-layer-python.yml (hunk 2 above)
Comment (ocelotl) [reply .github/workflows/release-layer-python.yml]: I merged 2 steps, the displaying of that environment variable needs to be in a separate step, if not its value is not shown in the logs.

**Review by tylerbenson — APPROVED** [review]

File: python/src/otel/otel_sdk/requirements-nodeps.txt (hunk 3 above)
Comment (ocelotl) [reply python/src/otel/otel_sdk/requirements-nodeps.txt:30]: I'll use only `1.21.0` as the version of the layer, we also use `0.42b0` for some packages that are not stable yet, but `1.21.0` is a sufficient id for a release.

**Pull request merged**